### PR TITLE
Add GET `/users` endpoint

### LINKED ISSUE #942: Create `/users` endpoint
Users are generally an internal entity right now but we have some need to expose the user ID for the current user in order to supply `createdBy` filters (see #939).

So, let's make a GET `/users` endpoint possibly with support for an `authenticationId` filter, and possibly with a special `/users/me` endpoint as well.

In terms of access control I could either:

1. make `/users` require the administrative role.
2. make `/users` require an auth context and then restrict the users returned based on auth context (e.g. if you aren't an admin the only user you can see is yourself)
3. do none of the above and just create a `/users/me` that returns your single authenticated user.

I'm planning to do both 2 + 3.

### PR DESCRIPTION
**Pull request by slifty**
This PR adds a new `/users` endpoint which will allow the front end to look up the id associated with their session.

I decided to implement this with auth context awareness off the bat, which means administrative users can actually view ALL users in the system if they want.

Resolves #942

### CODE REVIEW

**Comment by slifty** [comment]
I *think* all of the untested lines are of the category of "it's OK not to test them"

File: src/database/operations/load/loadUserBundle.ts
Comment (bickelj) [inline src/database/operations/load/loadUserBundle.ts:16]: Is this supposed to be a bundle of Proposal or of User?

File: src/database/operations/load/loadUserBundle.ts
Comment (bickelj) [inline src/database/operations/load/loadUserBundle.ts:26]: Same question here: `Proposal` or `User`?

**Review by reefdog — APPROVED** [review]
Functional review only: this works!

- ✅ Admin hitting `/users` got data about each user
- ✅ Admin hitting `/users?authenticationId={sub}` got data about that user (or an empty list)
- ✅ Non-admin hitting `/users` got data about only themselves
- ✅ Non-admin hitting `/users?authenticationId={sub}` got an empty list unless `{sub}` was themselves
- ✅ Public users got a 401.
  - I did notice the body of the 401 was an `UnknownError` and wasn't sure if that was correct, but I leave that up to you:
  ```json
  {
    "name": "UnknownError",
    "message": "No authorization token was found.",
    "details": [
      {}
    ]
  }
  ```

File: src/database/operations/load/loadUserBundle.ts
Comment (slifty) [reply src/database/operations/load/loadUserBundle.ts:16]: Great catch!  User!  This is why casting types can lead to bugs 😅

File: src/database/operations/load/loadUserBundle.ts
Comment (slifty) [reply src/database/operations/load/loadUserBundle.ts]: Fixed up!  Thank you.

File: src/queryParameters/extractAuthenticationIdParameters.ts
Comment (hminsky2002) [inline src/queryParameters/extractAuthenticationIdParameters.ts:22]: This is one of those type guards we were discussing correct? Very cool!

File: src/routers/usersRouter.ts
Comment (hminsky2002) [inline src/routers/usersRouter.ts:7]: All relevant boilerplate looks to have been added(handlers and routers)

**Review by hminsky2002 — COMMENTED** [review]
Looks good and now with fixed tests, going to approve after testing locally!

**Comment by hminsky2002** [comment]
Verifying success on all of the same checks as @reefdog . Hitting approve!

**Review by hminsky2002 — APPROVED** [review]

**Review by bickelj — APPROVED** [review]
I see the expected function signatures now, LGTM!

**Pull request merged**
